giskard/llm/client/__init__.py

```diff
@@ -1,14 +1,47 @@
+import os
+from enum import Enum
+
+from typing import Optional
+
 from .base import LLMClient, LLMFunctionCall, LLMOutput
 from .logger import LLMLogger
 
+
+class LLMClientAPI(Enum):
+    OPENAI = 1
+    AZURE = 2
+
+
 _default_client = None
+_default_llm_api: Optional[LLMClientAPI] = None
+_default_llm_model = os.getenv("GSK_LLM_MODEL", "gpt-4")
 
 
 def set_default_client(client: LLMClient):
     global _default_client
     _default_client = client
 
 
+def set_llm_api(llm__api: LLMClientAPI):
+    global _default_llm_api
+    _default_llm_api = llm__api
+
+
+def set_llm_model(llm_model: str):
+    global _default_llm_model
+    _default_llm_model = llm_model
+
+
+def get_default_llm_api() -> LLMClientAPI:
+    global _default_llm_api
+    if _default_llm_api is None:
+        _default_llm_api = os.getenv(
```

Comment:
I’m just realizing that `StrEnum` is only supported from Python 3.11. The code here would now work as `os.getenv` returns a string and we would need to compare.

Maybe I would entirely drop the enum for now, and just use pure strings, so that we don’t have to come up with magic ways to support 3.9 and 3.10.